### PR TITLE
Add Named Scopes & Collections example

### PR DESCRIPTION
**Pull request by maria-robobug**
https://issues.couchbase.com/browse/DOC-8163

This PR adds an example for named scopes and collections for KV operations.

I've also added a minor change in the Readme.

### CODE REVIEW

File: modules/devguide/examples/go/kv-collection-scope.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	// tag::namedcollectionupsert[]
+	agentScope := bucket.Scope("tenant_agent_00")
+	usersCollection := agentScope.Collection("users")
+
+	type userDoc struct {
+		Name           string `json:"name"`
+		PreferredEmail string `json:"preferred_email"`
+	}
+	document := userDoc{Name: "John Doe", PreferredEmail: "johndoe111@test123.test"}
```
Comment (osfameron) [inline modules/devguide/examples/go/kv-collection-scope.go:40]: formatting

File: modules/howtos/pages/kv-operations.adoc [hunk 2]
```diff
@@ -187,6 +187,21 @@ include::devguide:example$go/kv-counter.go[tag=decrement]
...
+
+It is possible to perform scoped key-value operations on named xref:7.0@server:learn:data/scopes-and-collections.adoc[`Collections`] _with the beta version of the next Couchbase Server release, 7.0β_.
+See the https://pkg.go.dev/github.com/couchbase/gocb/v2#Collection[API docs] for more information.
+
+CAUTION: This feature is marked xref:project-docs:compatibility.adoc#interface-stability[_Uncommitted_].
+Expect a promotion to _Committed_ API in a future minor release.
+
+Here is an example showing an upsert in the `users` collection, which lives in the `travel-sample.tenant_agent_00` keyspace:
+
+[source,go]
```
Comment (osfameron) [inline modules/howtos/pages/kv-operations.adoc]: Should be `[source,golang]` I think?

**Comment by osfameron** [comment]
`[source,golang]` as commented inline.
Otherwise LGTM 👍

File: modules/devguide/examples/go/kv-collection-scope.go (hunk 1 above)
Comment (maria-robobug) [reply modules/devguide/examples/go/kv-collection-scope.go:40]: Not sure what you mean here?
https://golang.org/cmd/gofmt/ 

gofmt seems happy with the formatting...

File: modules/howtos/pages/kv-operations.adoc (hunk 2 above)
Comment (maria-robobug) [reply modules/howtos/pages/kv-operations.adoc]: Hah! Good spot it should, will update this 👍

File: modules/devguide/examples/go/kv-collection-scope.go (hunk 1 above)
Comment (osfameron) [reply modules/devguide/examples/go/kv-collection-scope.go:40]: the data looked a bit awkward on one line, while the struct definition was spread out. But if gofmt insists on that then never mind :)

Comment (maria-robobug) [reply modules/devguide/examples/go/kv-collection-scope.go:40]: Ah ok, no worries, I getcha now! 
I guess if there were more fields I would probably not have it all in one line, but as the struct just has 2 it seemed ok to assign values that way.

**Review by RichardSmedley — APPROVED** [review]
:+1:

**Pull request merged**